tests/unit/test_compiler.py

```diff
@@ -417,3 +417,41 @@ def test_complex_grouping_ops_vs_nested_grouping_ops(
     )
 
     assert found_sql == expected_sql
+
+
+def test_label_compiler(faux_conn, metadata):
+    class CustomLower(sqlalchemy.sql.functions.FunctionElement):
+        name = "custom_lower"
+
+    @sqlalchemy.ext.compiler.compiles(CustomLower)
+    def compile_custom_intersect(element, compiler, **kwargs):
+        if compiler.dialect.name != "bigquery":
+            raise sqlalchemy.exc.CompileError(
+                f"custom_lower is not supported for dialect {compiler.dialect.name}"
+            )
```

Comment:
This is failing our coverage.

```
tests/unit/test_compiler.py                  169      1      2      1    99%   429
```

Since we don't expect to actually hit this line, I think a `# pragma: NO COVER` would be OK.

```suggestion
            # We only test with the BigQuery dialect, so this should never happen.
            raise sqlalchemy.exc.CompileError(  # pragma: NO COVER
                f"custom_lower is not supported for dialect {compiler.dialect.name}"
            )
```

Reply:
Thanks!